### PR TITLE
DOC-7628 scope/collection examples for Java Analytics

### PR DESCRIPTION
**Pull request by osfameron**
In particular I'm looking for comments on whether I've understood the Scoped dataset concept properly: this approach seems to be confirmed from http://review.couchbase.org/c/couchbase-jvm-clients/+/134947

> This adds analyticsQuery() methods to Collection. *In this implementation*,
> the dataverse name for the collection must be <bucket>.<scope> as that
> is what scope.analyticsQuery() uses for query_context

(Highlight of "In this implementation" is mine - is that an important caveat? Will this behaviour change/be implemented differently by other SDKs?)

NB: I've also made some comments about the Async/Reactive examples not running, and some warnings about thread termination. I think those are strictly speaking out of scope for this PR, but I'm happy to fix if anyone has suggestions!

### CODE REVIEW

File: modules/howtos/pages/analytics-using-sdk.adoc
Comment (RichardSmedley) [inline modules/howtos/pages/analytics-using-sdk.adoc:217]: Whilst _::./examples/Analytics.java_ will work,
_::example$Analytics.java_ is more idiomatically Antora, and enables Antora macros to do whatever they do should we happen to make more use of them ;)

**Review by RichardSmedley — APPROVED** [review]
LGTM, thanks @osfameron  - but I'll let @daschl comment on the Java before we merge.  :+1:

File: modules/howtos/examples/Analytics.java
Comment (daschl) [inline modules/howtos/examples/Analytics.java:180]: actually I'm not sure you need to do it like this way with 7.0 .. I think you can query collections directly with analytics now? (so no need to create a specific dataset for it)

**Review by daschl — CHANGES_REQUESTED** [review]
So I think while this approach works, you should be able to query the collection directly in 7.0, or am I mistaken?

File: modules/howtos/examples/Analytics.java
Comment (osfameron) [reply modules/howtos/examples/Analytics.java:180]: @daschl I couldn't get that to work either via the SDK or the workbench Analytics query -

e.g.  I get an error like “Cannot find dataset airline in dataverse travel-sample.inventory nor an alias with name airline!” if I try to run `select * from travel-sample.inventory.airline` (e.g. against a collection directly, instead of an analytics dataset)

So I'm very Confused! 😕 

Do you have a working simpler example, or other advice?

**Comment by osfameron** [comment]
This is subsumed by https://github.com/couchbase/docs-sdk-java/pull/214/files which is now merged.

**Pull request closed without merging**
